rust/toolchain.bzl

```diff
@@ -25,6 +25,13 @@ def _get_rustc_env(ctx):
     "CARGO_PKG_HOMEPAGE=",
   ]
 
+def _get_comp_mode_codegen_opts(ctx, toolchain):
+  comp_mode = ctx.var["COMPILATION_MODE"]
+  if not comp_mode in toolchain.compilation_mode_opts:
+    fail("Unrecognized compilation mode %s for toolchain." % comp_mode)
```

Comment:
@damienmg, I didn't see your target triple changes in so leaving that out of the message (for now).